src/core_plugins/kibana/public/context/components/size_picker/size_picker.js

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import contextSizePickerTemplate from './size_picker.html';
+import './size_picker.less';
+
+const module = uiModules.get('apps/context', [
+  'kibana',
+]);
+
+module.directive('contextSizePicker', function ContextSizePicker() {
+  return {
+    bindToController: true,
+    controller: ContextSizePickerController,
+    controllerAs: 'contextSizePicker',
+    replace: true,
+    restrict: 'E',
+    scope: {
+      count: '=',
+      setCount: '=',
+    },
+    template: contextSizePickerTemplate,
+  };
+});
+
+
+function ContextSizePickerController() {
+  this.getOrSetCount = (count) => (
+    _.isUndefined(count) ? this.count : this.setCount(count)
```

Comment:
Even though I was unable to actually produce an inconsistency like the one you are worried about, I have added code to properly call `$rollbackViewValue` when a value change is being propagated from the outside. I don't see how two watches would provide any advantage over that.